services/src/main/java/org/keycloak/broker/saml/SAMLEndpoint.java

```diff
@@ -782,4 +792,62 @@ private NameIDType getSubjectNameID(final AssertionType assertion) {
         SubjectType.STSubType subType = subject.getSubType();
         return subType != null ? (NameIDType) subType.getBaseID() : null;
     }
+
+    private boolean validateInResponseToAttribute(Element documentElement, Element assertionElement, String expectedRequestId) {
+        // If we are not expecting a request ID, don't bother
+        if (expectedRequestId == null || expectedRequestId.isEmpty())
+            return true; 
+
+        // We are expecting a request ID so we are in SP-initiated login, attribute InResponseTo must be present
+        if (!documentElement.hasAttribute("InResponseTo")) {
```

Comment:
Please use constants from `SAMLProtocolQNames` instead of the string literals

Reply:
Replaced in tests, no longer needed in main code as it's abstracted by using `responseType`. Thanks!